tendermint-lite/src/store.rs

```diff
@@ -25,17 +25,18 @@ impl MemStore {
     pub fn add(&mut self, trusted: State) -> Result<(), Error> {
         let height = trusted.last_header().header().height();
         self.height = height;
-        self.store.insert(height, trusted);
+        self.store
+            .insert(height, serde_json::to_string(&trusted).unwrap());
         Ok(())
     }
 
-    pub fn get(&self, h: Height) -> Result<&State, Error> {
+    pub fn get(&self, h: Height) -> Result<State, Error> {
         let mut height = h;
         if h == 0 {
             height = self.height
         }
         match self.store.get(&height) {
-            Some(state) => Ok(state),
+            Some(s) => Ok(serde_json::from_str(&s).unwrap()),
```

Comment:
What is the reason for serializing in the MemStore?

Reply:
It's only to demostrate it's serializable, but I'm ok to remove it.